### PR TITLE
Fix clippy lints for rlp-derive

### PR DESCRIPTION
**Pull request by vorot93**

### CODE REVIEW

File: rlp-derive/tests/rlp.rs [hunk 1]
```diff
@@ -10,62 +10,62 @@ use rlp::{decode, encode};
 use rlp_derive::{RlpDecodable, RlpDecodableWrapper, RlpEncodable, RlpEncodableWrapper};
 
 #[derive(Debug, PartialEq, RlpEncodable, RlpDecodable)]
-struct Foo {
+struct Item {
```
Comment (ordian) [inline rlp-derive/tests/rlp.rs]: lol, I didn't know it this lint existed https://rust-lang.github.io/rust-clippy/current/index.html#blacklisted_name
Not sure how useful it is though

File: rlp-derive/tests/rlp.rs [hunk 2]
```diff
@@ -10,62 +10,62 @@ use rlp::{decode, encode};
...
 		a: String,
 		/// It works with other attributes.
 		#[rlp(default)]
 		b: Option<Vec<u8>>,
 	}
 
-	let attack_of = String::from("clones");
-	let foo = Foo { a: attack_of.clone() };
+	let attack_of = "clones";
+	let item = Item { a: attack_of.into() };
```
Comment (ordian) [inline rlp-derive/tests/rlp.rs]: it was a pun on `attack of clones` :(

Comment (dvdplm) [reply rlp-derive/tests/rlp.rs]: Clippy has exactly zero sense of humour.

**Comment by dvdplm** [comment]
Can someone point me to the lint ihat replaces `match` with `if let … else`?

**Comment by niklasad1** [comment]
> Can someone point me to the lint ihat replaces match with if let … else?

https://rust-lang.github.io/rust-clippy/master/index.htm, `single_match_else`, I don't understand how to link to it :cold_sweat:

**Comment by dvdplm** [comment]
> I'm not a fan of this lint because it is more personal style preference, should be whitelisted IMHO.

I for one much prefer the `match`. (link is https://rust-lang.github.io/rust-clippy/master/#single_match_else btw, you click the paragraph-symbol, the backwards facing "P" thingy).

@vorot93 do you mind whitelisting this one and revert those changes?

**Comment by vorot93** [comment]
I can whitelist but I do think that `if let else` both reads much more naturally *and* reduces right shift.

**Comment by ordian** [comment]
Let's obey clippy.

**Pull request merged**